packages/@react-aria/radio/src/useRadio.ts

```diff
@@ -68,7 +68,14 @@ export function useRadio(props: AriaRadioProps, state: RadioGroupState, ref: Ref
   }), ref);
   let interactions = mergeProps(pressProps, focusableProps);
   let domProps = filterDOMProps(props, {labelable: true});
-  let tabIndex = state.lastFocusedValue === value || state.lastFocusedValue == null ? 0 : -1;
+  let tabIndex = -1;
+  if (state.selectedValue != null) {
+    if (state.selectedValue === value) {
+      tabIndex = 0;
+    }
+  } else if (state.lastFocusedValue === value || state.lastFocusedValue == null) {
+    tabIndex = 0;
+  }
```

Comment:
should we also accommodate for the case where the controlled value is set to a disabled radio? Probably tough to do here since we can't prevent that in the RadioGroup's state and each radio doesn't know if the selectedValue maybe disabled and thus can't set its tabIndex to 0 instead...

Reply:
The only way that could happen i think is if the app set the value to the disabled radio? Seems unlikely.